src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -399,7 +399,12 @@ public ResponseEntity<ProgrammingSubmission> lockAndGetProgrammingSubmission(@Pa
         // prepare programming submission for response
         programmingSubmissionService.hideDetails(programmingSubmission, user);
         // remove automatic results before sending to client
-        programmingSubmission.setResults(programmingSubmission.getManualResults());
+        if (correctionRound >= programmingSubmission.getManualResults().size()) {
+            programmingSubmission.setResults(List.of());
+        }
+        else {
+            programmingSubmission.setResults(List.of(programmingSubmission.getManualResults().get(correctionRound)));
+        }
```

Comment:
Maybe that's just personal preference, but I like to use these methods in such cases to signal the clear intention. With List.of() someone else could just add other elements to this list.
```suggestion
        if (correctionRound >= programmingSubmission.getManualResults().size()) {
            programmingSubmission.setResults(Collections.emptyList());
        }
        else {
            programmingSubmission.setResults(Collections.singletonList(programmingSubmission.getManualResults().get(correctionRound)));
        }
```

Reply:
Yes, sounds reasonable :+1: thank you Lucas